src/main/java/uk/gov/companieshouse/api/accounts/model/entity/AccountDataEntity.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.companieshouse.api.accounts.model.entity;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import javax.validation.constraints.NotNull;
+import org.springframework.data.mongodb.core.mapping.Field;
+import uk.gov.companieshouse.GenerateEtagUtil;
+import uk.gov.companieshouse.api.accounts.Kind;
+import uk.gov.companieshouse.api.accounts.LinkType;
+
+public class AccountDataEntity {
+
+    @Field("id")
+    private String id;
+
+    @NotNull
+    @Field("period_end_on")
+    private LocalDate periodEndOn;
+
+    private Map<String, String> links = new HashMap<>();
+
+    private String kind;
+
+    private String etag;
+
+    protected AccountDataEntity build(String id){
+        this.id = id;
+        this.etag = GenerateEtagUtil.generateEtag();
+        this.kind = Kind.ACCOUNT.getValue();
+
+        Map<String, String> links = new HashMap<>();
+        
+        links.put(LinkType.SELF.getLink(), "");
+        this.links = links;
+        return this;
+    }
```

Comment:
Yeh, that is a discussion me and Steve has had whereas we would use DTOs instead. The reason why the build method exists is because I thought DTOs weren't an option.

Reply:
Lets assume the design can change (because it can) - this seems to be something we should get right now as opposed to creating more work for ourselves later to rework

Reply:
We will raise another story for the DTOs because that in itself needs some investigation.